authentication/views.py

```diff
@@ -77,3 +93,40 @@ def post(self, request):
             print("Exception", e)
             return Response(status=status.HTTP_400_BAD_REQUEST)
 
+
+class PasswordRecoveryAPIView(APIView):
+    def post(self, request):
+        email = request.data.get('email')
+        try:
+            validate_email(email)
+        except ValidationError:
+            return Response({'error': 'Invalid email address'}, status=400)
+
+        try:
+            user = CustomUser.objects.get(email=email)
+        except CustomUser.DoesNotExist:
+            return Response({'error': 'User with this email does not exist'}, status=404)
+
+        jwt_token = Utils.generate_token(user.email, user.pk)
+        reset_link = f"{settings.FRONTEND_URL}/auth/password-reset/{jwt_token}/"
+        try:
+            Utils.send_password_reset_email(email, reset_link)
+        except Exception as e:
+            return Response({'error': 'Failed to send email'}, status=500)
+        return Response({'message': 'Password reset email sent successfully'}, status=status.HTTP_200_OK)
+
+
+
+class PasswordResetView(APIView):
+    serializer_class = PasswordRecoverySerializer
+
+    def post(self, request, jwt_token):
+        uid, email, exp = Utils.decode_token(jwt_token)
+        user = Utils.get_user(uid, email)
+        if user is not None:
+            serializer = self.serializer_class(instance=user, data=request.data)
+            serializer.is_valid(raise_exception=True)
+            serializer.save()
+            return JsonResponse({'message': 'Password reset successfully'}, status=status.HTTP_200_OK)
```

Comment:
Using status.HTTP_200_OK is appropriate here, but ensure consistency in how you reference status codes. In the email exception handling  (line 115), you used status=500 directly. It's better to use DRF's status constants (```status.HTTP_500_INTERNAL_SERVER_ERROR```) for readability and consistency

Reply:
Thank you!